### ISSUE TITLE
[Feature] Color Triggers

### ISSUE THREAD

**Issue by dak** [hankinsoft/MudClient#4]
Allow lines from the MUD to be re-stylized with a different color text (or even modify the text).

**Comment by hankinsoft** [comment]
So you want a function that lets you add/modify received text (with the ability to colorize it)?

Current with matches, I use parameters such as \0, \1 to get the match string, but I'm thinking I can add match objects with some basic psudocode like:

class match {
  str matchedString;
  int location;
  int length;
}

var matches = [];  // Populated via mud client

Then in javascript you could do something like:

replaceText(match[0].location, match[1].length, "Some new text");

And also allow the "Some new text" to contain 'color macros'.

Would something like that be what you are looking for?

**Comment by dak** [comment]
Yeah, that would be perfect. I was just aiming for colorization, but exposing a full replaceText along with colorization would be fantastic.

**Comment by hankinsoft** [comment]
I've submitted build 2026.09 (50), which includes the output-editing trigger functions:

gag() hides the matched line
highlight(foreground, background) recolours it (colour name, hex, or 256-colour palette index)
substitute(newText) replaces the matched text.

Documented here: https://www.mudclient.app/blog/19 (see "Output editing")

Once the build is available (usually within 24 hours), please give it a try and let me know if that sorts out what you want.
